### PR TITLE
added logic to consider locality synonyms on accommodation mapping

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Nakijin.Updater/Workers/ActualLocationsSearchHelper.cs [hunk 1]
```diff
@@ -126,6 +131,28 @@ private async Task InitializeCountryName(string countryCode)
...
+                select new
+                {
+                    CountryName = c.Names.En,
+                    SynonymName = lc.Names.En,
+                    LocalityId = l.Id,
+                    LocalityNames = l.Names
+                }).ToListAsync();
+
+            CountryLocalitiesBySynonyms = localitiesBySynonyms
+                .ToDictionary(ls => _locationNameNormalizer.GetNormalizedLocalityName(ls.CountryName, ls.SynonymName),
```
Comment (kirillta) [inline HappyTravel.Nakijin.Updater/Workers/ActualLocationsSearchHelper.cs:149]: Wouldn't that crash on similar keys?

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Updater/Workers/ActualLocationsSearchHelper.cs:149]: There must not be similar keys, the collection is only for one country.

**Review by kirillta — APPROVED** [review]

**Review by oleg-n-k — APPROVED** [review]

File: HappyTravel.Nakijin.Updater/Workers/Ancillary/InvalidAccommodationMappingAnalyzer.cs [hunk 2]
```diff
@@ -53,8 +61,7 @@ private void ClearInvalidAccommodationMappingsTable()
         }
 
 
-        private async Task WriteInvalidMappingToTheDb(
-            List<(int id, List<(Suppliers supplier, string supplierId)> suppliers, AccommodationKeyData keyData)> accommodationsData,
+        private async Task SaveInvalidMappings(List<(int id, List<(Suppliers supplier, string supplierId)> suppliers, AccommodationKeyData keyData)> accommodationsData,
```
Comment (kirillta) [inline HappyTravel.Nakijin.Updater/Workers/Ancillary/InvalidAccommodationMappingAnalyzer.cs:64]: Not clear, what `Save` means here

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Updater/Workers/Ancillary/InvalidAccommodationMappingAnalyzer.cs:64]: `Add` is more clear ?

**Pull request merged**